chain/consensus/mir/state_manager.go

```diff
@@ -137,6 +140,9 @@ func (sm *StateManager) RestoreState(checkpoint *checkpoint.StableCheckpoint) er
 
 		log.Infof("Restoring state from checkpoint at height: %d", ch.Height)
 
+		// Restore the height.
+		sm.height = ch.Height
```

Comment:
Note: this is the height of the chain (in Lotus terms) not Mir's epoch (from the logic it seems that it is correct, but just as a sanity-check)

Reply:
That is the approach that @matejpavlovic suggested to avoid calculating the height based on the previous epoch that could be removed.

Reply:
Gotcha! I'll review everything again, but I think this makes sense. Thanks!